Merge master into stable

mprahl opened the pull request.

juliusvonkohout committed:
chore: improved securitycontext for mysql (#11678)

* Update mysql-deployment.yaml

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

* Update mysql-deployment.yaml

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

* Update mysql-deployment.yaml

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

---------

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

mprahl committed:
Ignore empty PULL_NUMBER environment variables in SDK tests (#11714)

On pushes to the master branch, PULL_NUMBER was an empty string.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

chensun committed:
chore(sdk): Relax kfp-server-api upper bound to include 2.4 (#11685)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

chensun committed:
chore(sdk): release kfp sdk 2.12.1 (#11715)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

ntny committed:
fix(backend) fix execution-level retry on the Argo Workflows backend (#11673)

Signed-off-by: ntny <ntny1986@gmail.com>
Signed-off-by: arpechenin <arpechenin@avito.ru>
Co-authored-by: arpechenin <arpechenin@avito.ru>

HumairAK committed:
remove unused function (#11719)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

droctothorpe committed:
fix(frontend): compatibility with pod_names v1 (#11682)

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: cmdevoto <carolined321@gmail.com>

juliusvonkohout committed:
Chore: fix the PPC image for the dockerhub ratelimit and add the securitycontext (#11669)

* fix the PPC image

Signed-off-by: juliusvonkohout <45896133+juliusvonkohout@users.noreply.github.com>

* Update deployment.yaml

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

---------

Signed-off-by: juliusvonkohout <45896133+juliusvonkohout@users.noreply.github.com>
Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

HumairAK committed:
chore(release): bumped version to 2.4.1 (#11718)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

Commit:
feat(backend): Add types for KFP Kubernete Native API (#11672)

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

droctothorpe committed:
chore(frontend): update kubernetes client version to 0.12.1 (#11737)

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: ColeVogs <colevogs@gmail.com>

juliusvonkohout committed:
fix: minio fsgroup for popular clusters (#11734)

Signed-off-by: Julius von Kohout <45896133+juliusvonkohout@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/net from 0.33.0 to 0.36.0 in /test/tools/project-cleaner (#11748)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.33.0 to 0.36.0.
- [Commits](https://github.com/golang/net/compare/v0.33.0...v0.36.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

madmecodes committed:
Fix Istio sidecar injection by moving from annotations to labels (#11750)

Signed-off-by: madmecodes <ayushguptadev1@gmail.com>

droctothorpe committed:
docs(ui): add multi-user proxy instructions (#11743)

* docs(ui): add multi-user proxy instructions

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

* Update frontend/README.md

Co-authored-by: Anish Asthana <anishasthana1@gmail.com>
Signed-off-by: Alex <mythicalsunlight@gmail.com>

---------

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Signed-off-by: Alex <mythicalsunlight@gmail.com>
Co-authored-by: Anish Asthana <anishasthana1@gmail.com>

dependabot[bot] committed:
chore(deps): bump github.com/expr-lang/expr from 1.16.0 to 1.17.0 (#11756)

Bumps [github.com/expr-lang/expr](https://github.com/expr-lang/expr) from 1.16.0 to 1.17.0.
- [Release notes](https://github.com/expr-lang/expr/releases)
- [Commits](https://github.com/expr-lang/expr/compare/v1.16.0...v1.17.0)

---
updated-dependencies:
- dependency-name: github.com/expr-lang/expr
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

anishasthana committed:
chore(backend): Allow specification of image registry when building images (#11759)

Signed-off-by: Anish Asthana <anishasthana1@gmail.com>

stijntratsaertit committed:
fix(sdk): avoid conflicting component names in DAG when reusing pipelines (#11071)

* make component names in dag more unique

Signed-off-by: Stijn Tratsaert <stijn.tratsaert.it@gmail.com>

* tweak ordering so that renaming is handled in LIFO fashion

Signed-off-by: Stijn Tratsaert <stijn.tratsaert.it@gmail.com>

* add test that reuses a pipeline multiple times

Signed-off-by: Stijn Tratsaert <stijn.tratsaert.it@gmail.com>

---------

Signed-off-by: Stijn Tratsaert <stijn.tratsaert.it@gmail.com>

mprahl committed:
bug(backend,sdk): Use a valid path separator for Modelcar imports (#11767)

Forward slashes are invalid characters in a path and can't be escaped.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

Commit:
feat(proto): Add TTL fields to KFP IR yaml proto (#11758)

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

lociko committed:
fix(CI) Github action is vulnerable to code execution via `comment body` (#11772)

* Fix code execution on untrusted user input

Signed-off-by: VS <vspachyn@gmail.com>

* Remove redundant env variable 

Signed-off-by: VS <vspachyn@gmail.com>

---------

Signed-off-by: VS <vspachyn@gmail.com>

akagami-harsh committed:
Fix PSS restricted warnings (#11751)

* fix pss warnings

Signed-off-by: Harshvir Potpose <hpotpose62@gmail.com>

* fix

Signed-off-by: Harshvir Potpose <hpotpose62@gmail.com>

---------

Signed-off-by: Harshvir Potpose <hpotpose62@gmail.com>

hbelmiro committed:
test: Fixed and improved log collection (#11773)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

gmfrasca committed:
fix(backend): Include missing go.mod for cacheserver/viewercontroller images (#11776)

- Add api and k8s_platform go.mod files so cacheserver and
  viewercontroller images will build

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

hbelmiro committed:
test: Fixed `collect-logs.sh` path (#11775)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

HumairAK committed:
feat(backend/sdk): Add input parameterization for various k8s resources  (#11770)

* add backend support for k8s platform inputs

This change adds driver support for input parameter support for the
kubernetes platform spec. Input resolution change is extracted and made
more generic so it may be re-used when building out the container spec
for the k8s config.

Also add unit tests for constant & runtime input parameters.
TaskOutput parameter support are omitted due to a lack of appropriate
mlmd mock framework.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* add sdk implementation for k8s params inputs

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* add tests for k8s input params

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* add setup/teardown of prereqs and secret tests

and update/re-enable secret env tests

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* have kfp sample tests use local python pkgs

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* use a better configmap k8s name input...

Add support for multiple input types for pull secrets.
Clarify toleration docstring
Remove unnecessary resolve function

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

HumairAK committed:
chore(backend): refactor driver tests (#11777)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

mprahl committed:
feat(backend): Create a validating webhook for the PipelineVersion kind (#11774)

* Create a validating webhook for the PipelineVersion kind

Signed-off-by: VaniHaripriya <vmudadla@redhat.com>

* Fix the manifests for the validating webhook

This fixes deployment and local development for the Kubernetes native
API manifests.

This also addresses other feedback.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: VaniHaripriya <vmudadla@redhat.com>
Signed-off-by: mprahl <mprahl@users.noreply.github.com>
Co-authored-by: VaniHaripriya <vmudadla@redhat.com>

dependabot[bot] committed:
chore(deps): bump sagemaker from 2.1.0 to 2.237.3 in /samples/contrib/aws-samples/hosting_model_monitor_pipeline (#11769)

Bumps [sagemaker](https://github.com/aws/sagemaker-python-sdk) from 2.1.0 to 2.237.3.
- [Release notes](https://github.com/aws/sagemaker-python-sdk/releases)
- [Changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.md)
- [Commits](https://github.com/aws/sagemaker-python-sdk/compare/v2.1.0...v2.237.3)

---
updated-dependencies:
- dependency-name: sagemaker
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hbelmiro committed:
test: Added the result of `kubectl get pods` to logs when waiting for pods to get ready (#11779)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

dependabot[bot] committed:
chore(deps): bump @babel/helpers from 7.24.0 to 7.26.10 in /frontend (#11747)

Bumps [@babel/helpers](https://github.com/babel/babel/tree/HEAD/packages/babel-helpers) from 7.24.0 to 7.26.10.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.26.10/packages/babel-helpers)

---
updated-dependencies:
- dependency-name: "@babel/helpers"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

HumairAK committed:
chore(backend): upgade go and deps (#11780)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

Abhinav2777 committed:
fix(metadata-writer): use mlmd_store.get_context_types() instead of workaround (#11753)

Replaced the previous workaround for checking if the DB is empty with
mlmd_store.get_context_types(), as the upstream issue (#28) is now resolved.

Signed-off-by: Abhinav Sai D <abhinavsai491@tutanota.com>

hbelmiro committed:
chore(test): Replaced the unmaintained rerun action (#11784)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
chore(test): Bumped ubuntu runner to 24.04 in rerun-pr-jobs.yml (#11786)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

dependabot[bot] committed:
chore(deps): bump @babel/runtime and @material-ui/icons in /frontend (#11787)

Bumps [@babel/runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-runtime) to 7.27.0 and updates ancestor dependencies [@babel/runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-runtime) and [@material-ui/icons](https://github.com/mui-org/material-ui/tree/HEAD/packages/material-ui-icons). These dependencies need to be updated together.


Updates `@babel/runtime` from 7.14.6 to 7.27.0
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.0/packages/babel-runtime)

Updates `@babel/runtime` from 7.16.7 to 7.27.0
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.0/packages/babel-runtime)

Updates `@babel/runtime` from 7.2.0 to 7.27.0
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.0/packages/babel-runtime)

Updates `@babel/runtime` from 7.14.8 to 7.27.0
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.0/packages/babel-runtime)

Updates `@babel/runtime` from 7.14.0 to 7.27.0
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.0/packages/babel-runtime)

Updates `@babel/runtime` from 7.15.4 to 7.27.0
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.0/packages/babel-runtime)

Updates `@material-ui/icons` from 3.0.1 to 3.0.2
- [Release notes](https://github.com/mui-org/material-ui/releases)
- [Changelog](https://github.com/mui/material-ui/blob/v3.0.2/CHANGELOG.md)
- [Commits](https://github.com/mui-org/material-ui/commits/v3.0.2/packages/material-ui-icons)

---
updated-dependencies:
- dependency-name: "@babel/runtime"
  dependency-version: 7.27.0
  dependency-type: indirect
- dependency-name: "@babel/runtime"
  dependency-version: 7.27.0
  dependency-type: indirect
- dependency-name: "@babel/runtime"
  dependency-version: 7.27.0
  dependency-type: indirect
- dependency-name: "@babel/runtime"
  dependency-version: 7.27.0
  dependency-type: indirect
- dependency-name: "@babel/runtime"
  dependency-version: 7.27.0
  dependency-type: indirect
- dependency-name: "@babel/runtime"
  dependency-version: 7.27.0
  dependency-type: indirect
- dependency-name: "@material-ui/icons"
  dependency-version: 3.0.2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump axios from 1.6.0 to 1.8.2 in /frontend/server (#11732)

Bumps [axios](https://github.com/axios/axios) from 1.6.0 to 1.8.2.
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.6.0...v1.8.2)

---
updated-dependencies:
- dependency-name: axios
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gmfrasca committed:
chore: Upgrade Argo Workflows to v3.5.14, use Argo-provided images (#11783)

* chore: Upgrade Argo Workflows to v3.5.14

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* chore: Use upstream Argo Workflow images

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

---------

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

mprahl committed:
feat(backend): Add a mutating webhook for the PipelineVersion kind (#11782)

* Add a mutating webhook for the PipelineVersion kind

This mutating webhook adds labels to query pipeline versions by their
pipeline ID and pipeline name, and adds an owner's reference to the
pipeline it's a part of.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Delete the Argo Workflow object after successful SDK execution test

This will hopefully free up resources on the cluster and reduce the CI
flakes.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

HumairAK committed:
increase stale action timers (#11792)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

mprahl committed:
Fix recurring run output when always using latest (#11790)

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

parthea committed:
fix(sdk): allow google-cloud-storage < 4 (#11735)

Signed-off-by: Anthonios Partheniou <partheniou@google.com>

Commit:
fix(local): warn about oci:// not supported too (#11794)

AFAIU no remote schema is currently supported for Local. This patch
consolidates all possible values under an Enum and then checks the
supplied url against all possible prefixes.

Signed-off-by: Ihar Hrachyshka <ihar.hrachyshka@gmail.com>

milinddethe15 committed:
fix(deployment): Update kustomize manifests to use new label and patch syntax (#11733)

Signed-off-by: milinddethe15 <milinddethe15@gmail.com>

mprahl committed:
Handle optional pipeline inputs in the driver (#11788)

If the pipeline run is submitted without specifying an optional
parameter and there is no default, it was not handled by the driver.
The approach taken is explicitly set null for these values and let the
driver handle if the component parameter has a default that can be
used in the launcher.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

hbelmiro committed:
chore(test): Removed the no longer needed rerun-pr-jobs.yml (#11796)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

sradc committed:
docs: mention that `set_container_image` works with dynamic images (#11795)

Signed-off-by: Sidney Radcliffe <sidneyradcliffe@gmail.com>

HumairAK committed:
fix setup env for kfp k8s lib tests (#11798)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

droctothorpe committed:
feat(backend): implement logs as artifacts (#11762)

* feat(backend): implement logs as artifacts

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: cmdevoto <carolined321@gmail.com>

* Address feedback and update golangci-lint

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

* Implement flag

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: carter.fendley <carter.fendley@gmail.com>

---------

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: cmdevoto <carolined321@gmail.com>
Co-authored-by: carter.fendley <carter.fendley@gmail.com>

HumairAK committed:
(chore): add image builds for default branch (#11800)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

co63oc committed:
chore: Fix typos (#11668)

Signed-off-by: co63oc <co63oc@users.noreply.github.com>

droctothorpe committed:
Revert "feat(backend): implement logs as artifacts (#11762)" (#11807)

This reverts commit cd3e747b5de3d7e1e338e309cc57311dd4a91258.

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

hbelmiro committed:
feat(backend): Add the ability to set a proxy for accessing external resources (#11771)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

HumairAK committed:
chore(backend): removed tekton backend (#11813)

As per community discussion, the tekton backend is being removed due to
lack of maintainers.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

strangiato committed:
feat(sdk): add upload pipeline and upload pipeline version from pipeline function (#11804)

Signed-off-by: Trevor Royer <troyer@redhat.com>

HumairAK committed:
update driver & launcher image handling (#11533)

This change relies on manifest yamls to specify the launcher & driver
that is pinned to a specific KFP version. The goal is to decouple having
to build launcher/driver at separate stages compared to api server.

This is accomplished by setting the hardcoded default to point to
"latest" and during release, api server is built with this hardcoding,
and the images for driver/launcher are patched into manifests post build
along with the other images.

The apiserver deployment manifest is reformatted using yq so the next
time the release.sh is ran, the user is not surprised by the entire file
reformatting unexpectedly.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

# Conflicts:
#	manifests/kustomize/base/pipeline/ml-pipeline-apiserver-deployment.yaml

# Conflicts:
#	.github/resources/manifests/tekton/kustomization.yaml

droctothorpe committed:
feat(backend): implement logs as artifacts + CI updates (#11809)

* feat(backend): implement logs as artifacts

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: cmdevoto <carolined321@gmail.com>

* Address feedback and update golangci-lint

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

* Implement flag

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: carter.fendley <carter.fendley@gmail.com>

* Handle logs in kubernetesPlatformOps

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

* Broaden sdk execution test filter

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

* Delete MinIO PVC at end of SDK execution tests

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

* Emit more logs from sdk execution tests

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

---------

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: cmdevoto <carolined321@gmail.com>
Co-authored-by: carter.fendley <carter.fendley@gmail.com>

droctothorpe committed:
chore(tests): remove unused logic (#11816)

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

droctothorpe committed:
fix(tests): free up space in some test runners (#11818)

* fix(tests): free up space in some test runners

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

* Add comments and logs

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

---------

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>

mprahl committed:
fix(docs): Use the latest driver and launcher images in the dev environment (#11820)

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

mprahl committed:
fix(backend): Fix run submissions with OwnerReferencesPermissionEnforcement on (#11821)

If the run is associated with a ScheduledWorkflow and the
OwnerReferencesPermissionEnforcement admission controller is on, it
requires the update verb on scheduledworkflows/finalizers to set the
proper owner reference.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump @babel/runtime-corejs3 from 7.12.5 to 7.27.0 in /frontend (#11815)

Bumps [@babel/runtime-corejs3](https://github.com/babel/babel/tree/HEAD/packages/babel-runtime-corejs3) from 7.12.5 to 7.27.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.0/packages/babel-runtime-corejs3)

---
updated-dependencies:
- dependency-name: "@babel/runtime-corejs3"
  dependency-version: 7.27.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
fix(components): Set tensorboard_experiment_name to random uuid when uploading plots in Starry Net pipeline
fix(components): Upgrade default Tensorflow version when setting test set and retrieving training artifacts to random uuid when uploading plots in Starry Net pipeline
fix(components): Bump Starry Net dataprep, trainer and predictor image versions.

Signed-off-by: Googler <nobody@google.com>
PiperOrigin-RevId: 747428987

HumairAK committed:
(chore): add cleanup resources to sdk execution tests (#11823)

also disable argo logs

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

dolfim-ibm committed:
fix(deps): widen urllib3 upper bound to <3.0.0 (#11819)

Signed-off-by: Michele Dolfi <dol@zurich.ibm.com>

alyssacgoins committed:
fix(docs): Remove Podman as backend README pre-req (#11824)

* Remove Podman as backend README pre-req

Signed-off-by: agoins <alyssacgoins@gmail.com>

* Note Docker req for dev-kinda-cluster

Signed-off-by: agoins <alyssacgoins@gmail.com>

* Change back to Docker or Podman

Signed-off-by: agoins <alyssacgoins@gmail.com>

---------

Signed-off-by: agoins <alyssacgoins@gmail.com>

hbelmiro committed:
chore(test): Improved path filtering in the CI (#11806)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

HumairAK committed:
chore: update release notes (#11827)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

Commit:
chore: Add the OpenSSF Best Practices badge (#11543)

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

ntny committed:
fix component retry test (#11836)

Signed-off-by: ntny <ntny1986@gmail.com>
Signed-off-by: arpechenin <arpechenin@avito.ru>
Co-authored-by: arpechenin <arpechenin@avito.ru>

dependabot[bot] committed:
chore(deps): bump golang.org/x/net from 0.36.0 to 0.38.0 in /test/tools/project-cleaner (#11835)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.36.0 to 0.38.0.
- [Commits](https://github.com/golang/net/compare/v0.36.0...v0.38.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-version: 0.38.0
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hbelmiro committed:
chore(backend): Optimized Dockerfiles (#11834)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/net from 0.36.0 to 0.38.0 (#11838)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.36.0 to 0.38.0.
- [Commits](https://github.com/golang/net/compare/v0.36.0...v0.38.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-version: 0.38.0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mprahl committed:
fix(sdk): Add SDK support for setting resource limits on older KFP versions (#11839)

For context, the commit 70aaf8a9a469607dc6e4aad58d40b39c75363b99 removed
support for the old fields (without a resource_ prefix). This was
added back in commit 6ebf4aae0335424d3bc88175fd06a2b2ba05251f but done in a way
that broke any usage of pipeline input parameters but was to support the
current KFP backend which did not yet support the new fields. In commit
7c931ae20197b2309d7a8462f6ce099882a8f915, the old fields were removed
again but added support for the new field in KFP backend.

This commit addresses the case where a user is using a new SDK but with
a KFP backend prior to 2.4.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

HumairAK committed:
add missing release note updates to sdk main branch (#11842)

* add missing release note for #11839

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* add missing release note for #11804

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

hbelmiro committed:
fix(backend): fixed Dockerfile (#11841)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

HumairAK committed:
chore(sdk): update go toolchain patch (#11845)

and also add api reference to k8s platform reference to kfp sdk api

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump minimist from 1.2.5 to 1.2.8 in /frontend (#11846)

Bumps [minimist](https://github.com/minimistjs/minimist) from 1.2.5 to 1.2.8.
- [Changelog](https://github.com/minimistjs/minimist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/minimistjs/minimist/compare/v1.2.5...v1.2.8)

---
updated-dependencies:
- dependency-name: minimist
  dependency-version: 1.2.8
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump torch from 1.0.0 to 2.6.0 in /samples/contrib/ibm-samples/ffdl-seldon/source/seldon-pytorch-serving-image (#11848)

Bumps [torch](https://github.com/pytorch/pytorch) from 1.0.0 to 2.6.0.
- [Release notes](https://github.com/pytorch/pytorch/releases)
- [Changelog](https://github.com/pytorch/pytorch/blob/main/RELEASE.md)
- [Commits](https://github.com/pytorch/pytorch/compare/v1.0.0...v2.6.0)

---
updated-dependencies:
- dependency-name: torch
  dependency-version: 2.6.0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump sagemaker from 2.1.0 to 2.237.3 in /components/aws/sagemaker/TrainingJob/samples/mnist-kmeans-training (#11847)

Bumps [sagemaker](https://github.com/aws/sagemaker-python-sdk) from 2.1.0 to 2.237.3.
- [Release notes](https://github.com/aws/sagemaker-python-sdk/releases)
- [Changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.md)
- [Commits](https://github.com/aws/sagemaker-python-sdk/compare/v2.1.0...v2.237.3)

---
updated-dependencies:
- dependency-name: sagemaker
  dependency-version: 2.237.3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hbelmiro committed:
chore(backend): fixed support for Podman in Makefile (#11844)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

HumairAK committed:
chore(sdk): release kfp sdk 2.12.2 (#11843)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

HumairAK committed:
add backend support for toleration lists.

clarify toleration json docs

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

HumairAK committed:
switch selenium image to ghcr

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

HumairAK committed:
add list or dict support for add toleration json

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

Commit:
Fix Integration tests

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

HumairAK committed:
chore: fix kfp-kubernetes pipeline spec resolution errors (#11868)

* use packaged pipeline spec in k8s platform pkg

kfp-kubernetes expects pipeline spec to be present in global namespace
this change updates generate proto code to replace this import to
correctly reference the spec in the kfp.pipeline_spec namespace.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* update to protoc 3.20

upgrading to 3.20 resolves issues where some of the proto python
generated code attempts to directly create descriptors. This issue is
encountered when trying to use kfp-kubernetes packages with kfp sdk,
forcing the user to downgrade protobuf to 3.20 or lower despite the
kfp sdk and kfp-kubernetes supporting >=4.0. This change resolves
this issue.

updating protoc also requires regeneration of all go and python proto code, as well us updates to the api generate and release images.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

HumairAK committed:
chore(release): bumped version to 2.5.0

HumairAK committed:
chore(sdk): kfp and kfp-kubernetes 2.13 & 1.5 (#11870)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

HumairAK committed:
update release doc paths & make script executable (#11871)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

chensun committed:
chore(components): GCPC 2.20.0 Release

Signed-off-by: Chen Sun <chesu@google.com>
PiperOrigin-RevId: 752751814

chensun committed:
chore(components): Relax KFP SDK dependency for GCPC

Signed-off-by: Chen Sun <chesu@google.com>
PiperOrigin-RevId: 752807981

HumairAK committed:
chore: bump master to release 2.5 (#11872)

* chore(release): bumped version to 2.5.0

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* remove unneeded ci files

these are added from the release script, until the scripts are updated,
these are manually removed.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* revert incorrect changelog changes for 2.5

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* correct 2.5.x change log (#11878)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump sagemaker from 2.1.0 to 2.237.3 in /components/aws/sagemaker (#11876)

Bumps [sagemaker](https://github.com/aws/sagemaker-python-sdk) from 2.1.0 to 2.237.3.
- [Release notes](https://github.com/aws/sagemaker-python-sdk/releases)
- [Changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.md)
- [Commits](https://github.com/aws/sagemaker-python-sdk/compare/v2.1.0...v2.237.3)

---
updated-dependencies:
- dependency-name: sagemaker
  dependency-version: 2.237.3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

h-joo committed:
No public description

Tested:
Signed-off-by: Hana Joo <hanajoo@google.com>
PiperOrigin-RevId: 753224247

mprahl committed:
feat(backend): Add the Kubernetes native pipeline store (#11881)

* Add the Kubernetes native pipeline store

This also improves cache update race conditions in the webhooks.

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

* Use controller-runtime for the non-caching client

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Put the Kubernetes native CI manifests under the Argo manifests

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Fix the flaky Kubernetes pipeline store tests

Some tests set the viper configuration of POD_NAMESPACE while others
didn't and so the order of the tests mattered. This now sets and resets
the viper configuration for each test.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Modify the suggested pipeline version name to be a valid K8s name

This is more important for the Kubernetes pipeline store.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Fix the ml-pipeline Service ports in K8s native mode

The KFP UI automatically uses the first port listed in the ml-pipeline
Service to communicate with the KFP API. Using a JSON patch to add the
webhook port ensures it doesn't change the order.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>
Signed-off-by: mprahl <mprahl@users.noreply.github.com>
Co-authored-by: Ricardo M. Oliveira <rmartine@redhat.com>

chjdev committed:
fix(sdk): resolve issue when creating pipeline version from pipeline name using the cli. Fixes #11810 (#11866)

It is checked that not both are passed in as parameters, however in the
case of passing in the `pipeline_name` the `pipeline_id` is fetched via
the client and then both values are passed to `upload_pipeline_version`
defeating the initial check and raising an error.

Simply removing the `pipeline_name` argument from the call to
`upload_pipeline_version` fixes this as the `pipeline_id` is always set.

Signed-off-by: Christian Junker <chjdev@users.noreply.github.com>

mprahl committed:
fix(backend): Stop logging the strack trace on benign user errors (#11883)

Prior to this change, if a user requested a pipeline version that no
longer exists, it'd cause a whole stack trace to be displayed in the
logs. For benign errors, this now is a single info log.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

HumairAK committed:
chore: update all owners files (#11886)

Various owners' files reviewers/approvers are no longer involved with
the community. Additionaly, we have various other folks that are
involved and have shown an active interest in reviewing various portions
of the code base. This change updates all owners files to reflect this
current state of the community.

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

mprahl committed:
feat(backend): Allow the launcher command to be configurable (#11888)

This allows the launcher command to be overridden with
the V2_LAUNCHER_COMMAND environment variable. This is useful if you need
to override the command to launch Delve for debugging or you have a
situation that requires using a different binary in the container image
based on the environment.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

zazulam committed:
feat(backend/sdk): enable dsl.Collected for parameters & artifacts (#11725)

* feat(backend/sdk): enable dsl.Collected for params & artifacts

Signed-off-by: zazulam <m.zazula@gmail.com>

* feat(backend): collect through loops & dags

Signed-off-by: zazulam <m.zazula@gmail.com>

To enable users to use loops similar to subdags, the initial collecting
implementation went only 1 layer deep of loops/subdags. This
implementation serves to handle multifacted approaches of pipelines that
users can generate.

---------

Signed-off-by: zazulam <m.zazula@gmail.com>

dandawg committed:
fix(sdk): fix pip install for dev (#11891)

Signed-off-by: Daniel Dowler <12484302+dandawg@users.noreply.github.com>

dandawg committed:
fix(sdk): fix pip install for dev (#11891)

Signed-off-by: Daniel Dowler <12484302+dandawg@users.noreply.github.com>
(cherry picked from commit 4503eae10d19518bc35310c57d2ef957b31f959c)

dandawg committed:
UPSTREAM: 11891: fix(sdk): fix pip install for dev

Signed-off-by: Daniel Dowler <12484302+dandawg@users.noreply.github.com>
(cherry picked from commit 4503eae10d19518bc35310c57d2ef957b31f959c)

franciscojavierarceo committed:
chore: Adding Adopters file for CNCF graduation (#11894)

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>

alyssacgoins committed:
feat(backend): parameterize retryStrategy  input in Argo workflow (#11861)

Prior to this commit, there was a workflow template added for every unique
retryStrategy setting. This is now consolidates to a single retryStrategy
template.

Signed-off-by: agoins <alyssacgoins@gmail.com>

mprahl committed:
chore: Add mprahl to the samples owners file (#11897)

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

HumairAK committed:
chore: handle empty tolerations dict/lists parrameterization  (#11898)

* remove unnecessary nil err

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

* ignore empty tolerations

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

---------

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

hbelmiro committed:
feat(backend): add the option to enable/disable cache globally (#11831)

* feat(backend): add the option to enable/disable cache globally

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* chore(backend): added logging when cache is disabled globally

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* Update backend/src/apiserver/resource/resource_manager.go

Co-authored-by: Giulio Frasca <giulio.m.frasca@gmail.com>
Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* chore(backend): Fixed Scheduled Workflows

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* chrore(backend): Added integration tests

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* chrore(backend): Removed container arg when --cache_enabled is true

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* Update backend/src/v2/cmd/driver/main.go

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* Update backend/src/v2/cmd/compiler/main.go

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* chrore(backend): Renamed to CacheDisabled and removed pointers

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

* chrore(backend): Added Argo compiler unit test with the cache disabled

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

---------

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>
Co-authored-by: Giulio Frasca <giulio.m.frasca@gmail.com>
Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>

kaikaila committed:
feat(frontend): Add "Always Use Latest Version" option for recurring runs (fixes #11581) (#11755)

Implements a checkbox UI component that enables automatic pipeline version selection for recurring runs, building upon @muzzlol's original work.

Key changes:

- Added "Use Latest Version" toggle functionality
- Removed localStorage dependency (per @mprahl's feedback)
- Improved version selector accessibility labels
- Added comprehensive test coverage

Signed-off-by: kaikaila <lyk2772@126.com>
Co-authored-by: muzzlol <muzxmmilkhxn@gmail.com>

mprahl committed:
Add the stable and rhoai branches to the CI

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

(cherry picked from commit df01083cd2175829e736f8e098aadc975e9bb092)
Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

mprahl committed:
Add the stable and rhoai branches to the CI

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

HumairAK committed:
add maintainer to kfp (#11900)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

Shekharrajak committed:
chore: correct broken links in README.md  (#11896)

* Update README.md

Signed-off-by: Shekhar Prasad Rajak <5774448+Shekharrajak@users.noreply.github.com>

* Part 3 link reverted

Co-authored-by: Helber Belmiro <helber.belmiro@gmail.com>
Signed-off-by: Shekhar Prasad Rajak <5774448+Shekharrajak@users.noreply.github.com>

---------

Signed-off-by: Shekhar Prasad Rajak <5774448+Shekharrajak@users.noreply.github.com>
Co-authored-by: Helber Belmiro <helber.belmiro@gmail.com>

mai-nakagawa committed:
fix(sdk): Resolves issue when using ParallelFor with param and depending tasks (#11903)

Signed-off-by: Mai Nakagawa <nakagawa.mai@gmail.com>

chensun committed:
chore(components): fix v1 model_evaluations components mingled under the preview module

Signed-off-by: Chen Sun <chesu@google.com>
PiperOrigin-RevId: 756808592

VaniHaripriya committed:
feat(backend): Add migration script to create Pipeline and PipelineVersion objects from the database to Kubernetes API (#11884)

Signed-off-by: VaniHaripriya <vmudadla@redhat.com>

hbelmiro committed:
Rebase to KFP 2.5.0

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
UPSTREAM: <carry>: Fixed Dockerfile

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

chensun committed:
chore(components): GCPC 2.20.1 Release

Signed-off-by: Chen Sun <chesu@google.com>
PiperOrigin-RevId: 757922879

anishasthana committed:
chore: Add IBM Research Foundation Model Data Engineering team (#11905)

Signed-off-by: Anish Asthana <anishasthana1@gmail.com>

hbelmiro committed:
UPSTREAM: 11911: chore(tests): fix KFP SDK tests

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
chore(tests): fix KFP SDK tests

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
chore(tests): fix KFP SDK tests (#11911)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

HumairAK committed:
chore(backend): break up driver logic (#11885)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

hbelmiro committed:
UPSTREAM: <carry>: Removed import

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

chensun committed:
chore(components): Update OS packages in GCPC container image

Signed-off-by: Chen Sun <chesu@google.com>
PiperOrigin-RevId: 758746089

openshift-merge-bot[bot] committed:
Merge pull request #171 from hbelmiro/test-ci

UPSTREAM: 11891, 11911: Fix SDK unit tests

VaniHaripriya committed:
feat(backend): Add CLI flags to support Kubernetes native API implementation (#11907)

Signed-off-by: VaniHaripriya <vmudadla@redhat.com>

hbelmiro committed:
chore(test): Update cache test timeout and polling intervals (#11916)

Adjusted the timeout from 3 to 4 minutes and polling interval from 10 to 5 seconds in the cache integration test. This ensures more robust testing by allowing sufficient time for operations to complete while increasing polling frequency.

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
UPSTREAM: <carry>: Add CA bundle configuration to container templates

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

mprahl committed:
docs(KEP): Propose a new pipeline run workspace feature (#11875)

* Propose a new pipeline run workspace feature

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Fix update-toc.sh and fix outdated ToCs

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

HumairAK committed:
chore(backend): upgrade mysql to 8.4 (#11917)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

cmdevoto committed:
chore: Enable go fmt as a lint check for Go code (#11830)

Signed-off-by: Caroline DeVoto <cmdevoto@users.noreply.github.com>
Co-authored-by: Caroline DeVoto <cmdevoto@users.noreply.github.com>

hbelmiro committed:
chore(test): fixed log collection in tests (#11910)

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

VaniHaripriya committed:
feat(proto): Add WorkspaceConfig and KubernetesWorkspaceConfig message types to pipeline_spec proto (#11921)

Signed-off-by: VaniHaripriya <vmudadla@redhat.com>

ntny committed:
- fix(launcher): missing executorInput parameter values caused by {{$}} evaluation order (#11925)

Signed-off-by: arpechenin <arpechenin@avito.ru>
Co-authored-by: arpechenin <arpechenin@avito.ru>

lekaf974 committed:
chore(ci): clean up unnecessary parentheses and fix the formatting of the CI files (#11926)

* chore(ci): clean up unnecessary parentheses

Signed-off-by: lekaf974 <matthieu.evrin@gmail.com>

* chore(ci): update adding missing spaces based on suggestions from code review

Co-authored-by: Helber Belmiro <helber.belmiro@gmail.com>
Signed-off-by: Matthieu Evrin <6630779+lekaf974@users.noreply.github.com>

---------

Signed-off-by: lekaf974 <matthieu.evrin@gmail.com>
Signed-off-by: Matthieu Evrin <6630779+lekaf974@users.noreply.github.com>
Co-authored-by: Helber Belmiro <helber.belmiro@gmail.com>

HumairAK committed:
docs(frontend): add ui dev docs (#11931)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

HumairAK committed:
feat: add openshift env to manifests (#11932)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

franciscojavierarceo committed:
chore: Updating KFP docs (#11927)

* chore: Updating KFP docs

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>

* remove html reference

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>

* updating python version

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>

* Update index.rst

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Francisco Arceo <farceo@redhat.com>

* Update quickstart.rst

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Francisco Arceo <farceo@redhat.com>

* Update quickstart.rst

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Francisco Arceo <farceo@redhat.com>

* rebasing

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>

* removed overview file

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>

* updated to make things look better

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>

* Update docs/source/overview.rst

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Francisco Arceo <farceo@redhat.com>

* Update docs/source/overview.rst

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Francisco Arceo <farceo@redhat.com>

* Update docs/source/installation.rst

Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>
Signed-off-by: Francisco Arceo <farceo@redhat.com>

* updating conf.py

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>

---------

Signed-off-by: Francisco Javier Arceo <farceo@redhat.com>
Signed-off-by: Francisco Arceo <farceo@redhat.com>
Co-authored-by: Matt Prahl <mprahl@users.noreply.github.com>

HumairAK committed:
Merge pull request #170 from hbelmiro/rebase-2.5.0

Rebase 2.5.0

alyssacgoins committed:
docs: add GoLand-specific configs to backend Readme (#11919)

Updates V2_DRIVER_COMMAND argument and adds runtime arguments for starting a remote debug session in GoLand.

Signed-off-by: agoins <alyssacgoins@gmail.com>

mprahl committed:
test: Update the Kubernetes and Python version ranges in the CI (#11924)

* Minimize the Kubernetes version range in the CI

This reduces the matrix to only include the low and high versions.

See the KFP community call notes for more context:
https://docs.google.com/document/d/1cHAdK1FoGEbuQ-Rl6adBDL5W2YpDiUbnMLIwmoXBoAU/edit?tab=t.0#heading=h.fovolzywu84d

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Reduce the testing matrix for Python versions

This reduces the Python versions being tested to the low and high
versions to reduce GitHub CI consumption.

See the KFP community call discussion for more context:
https://docs.google.com/document/d/1cHAdK1FoGEbuQ-Rl6adBDL5W2YpDiUbnMLIwmoXBoAU/edit?tab=t.0#heading=h.fovolzywu84d

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Add an option to run integration tests locally

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Stop using whalesay in the tests

The container image is over 10 years old and is in a format that is
deprecated on newer Kubernetes versions.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

alyssacgoins committed:
feat(backend) implement retryStrategy for nested pipelines  (#11908)

Signed-off-by: agoins <alyssacgoins@gmail.com>

dependabot[bot] committed:
chore(deps): bump tornado from 6.4.2 to 6.5.1 in /test/sample-test (#11936)

Bumps [tornado](https://github.com/tornadoweb/tornado) from 6.4.2 to 6.5.1.
- [Changelog](https://github.com/tornadoweb/tornado/blob/master/docs/releases.rst)
- [Commits](https://github.com/tornadoweb/tornado/compare/v6.4.2...v6.5.1)

---
updated-dependencies:
- dependency-name: tornado
  dependency-version: 6.5.1
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alyssacgoins committed:
fix(tests): free up space in kfp samples test workflow (#11942)

Fixes flaky test behavior in kfp-samples.

Signed-off-by: agoins <alyssacgoins@gmail.com>

HumairAK committed:
chore: pipeline api spec minor release (#11939)

Signed-off-by: Humair Khan <HumairAK@users.noreply.github.com>

hbelmiro committed:
Rebase on top of https://github.com/kubeflow/pipelines/commit/4c966f77732c96784619506cf6e6e7211c364257

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
Fixed tests after rebase

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
Fixed OWNERS files after rebase

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
Fixed URLs

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

hbelmiro committed:
Fixed formatting

Signed-off-by: Helber Belmiro <helber.belmiro@gmail.com>

openshift-merge-bot[bot] committed:
Merge pull request #176 from hbelmiro/rebase-master

Rebase over kubeflow@4c966f7

mprahl committed:
feat(backend/frontend): Add the name field for pipelines and pipeline versions (#11952)

* Add the name field for pipelines and pipeline versions

The API only accepted display_name, which accepted arbitrary characters
but this does not work well in Kubernetes native API mode (Kubernetes
pipeline store) since object names must be a valid DNS subdomain.

This commit adds the name field which is required, but for backwards
compatibility, the API will accept just name or display_name and
duplicate the set value to the unset field. For existing users, they
will only see that there is an extra field of "name" on pipelines and
pipeline versions that is the same value as "display_name".

The database migration logic adds a new DisplayName column and sets the
values to what was in the Name column. Confusingly, the API field
"display_name" used to map the a database column of "Name".

The UI now handles both name and display name but for a simpler user
experience, the display name form fields when uploading a pipeline are
hidden when not in Kubernetes native API mode (Kubernetes pipeline
store) since there is no need to have a display name since there is no
naming restrictions on the name.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

* Add a better API error when a Kubernetes object name is invalid

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

---------

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

openshift-merge-bot[bot] committed:
Merge pull request #178 from mprahl/cherry-pick-display-name

Cherry-pick #11952: Add the name field for pipelines and pipeline versions

mprahl commented:
/override "Code scanning results"

The failure in the SDK which we don't ship.

mprahl commented:
/override "CodeQL"

The failure in the SDK which we don't ship.

dsp-developers commented:
A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-179`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-179`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-179`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-179`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-179`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-179`

dsp-developers commented:
An OCP cluster where you are logged in as cluster admin is required.

The Data Science Pipelines team recommends testing this using the Data Science Pipelines Operator.  Check [here](https://github.com/opendatahub-io/data-science-pipelines-operator) for more information on using the DSPO.

To use and deploy a DSP stack with these images (assuming the DSPO is deployed), first save the following YAML to a file named `dspa.pr-179.yaml`:
```yaml
apiVersion: datasciencepipelinesapplications.opendatahub.io/v1alpha1
kind: DataSciencePipelinesApplication
metadata:
  name: pr-179
spec:
  dspVersion: v2
  apiServer:
    image: "quay.io/opendatahub/ds-pipelines-api-server:pr-179"
    argoDriverImage: "quay.io/opendatahub/ds-pipelines-driver:pr-179"
    argoLauncherImage: "quay.io/opendatahub/ds-pipelines-launcher:pr-179"
  persistenceAgent:
    image: "quay.io/opendatahub/ds-pipelines-persistenceagent:pr-179"
  scheduledWorkflow:
    image: "quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-179"
  mlmd:  
    deploy: true  # Optional component
    grpc:
      image: "quay.io/opendatahub/mlmd-grpc-server:latest"
    envoy:
      image: "registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2"
  mlpipelineUI:
    deploy: true  # Optional component 
    image: "quay.io/opendatahub/ds-pipelines-frontend:pr-179"
  objectStorage:
    minio:
      deploy: true
      image: 'quay.io/opendatahub/minio:RELEASE.2019-08-14T20-37-41Z-license-compliance'
```

Then run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines.git
cd data-science-pipelines/
git fetch origin pull/179/head
git checkout -b pullrequest 06726057c1f0eb6603ce420ed12ecd574621cd4b
oc apply -f dspa.pr-179.yaml
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

mprahl committed:
fix(backend): Fix the pipeline samples (#11967)

After making name and display_name separate fields in the API, the
pipeline sample upload code needed to accomodate that, otherwise it
would try to upload the pipeline without a display name, which is a
required field. The backwards compatibility was handled at the API
server level but this code bypassed the API server endpoints and used
the resource manager directly.

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
(cherry picked from commit b4772693ae0f7d90425f604ebdafbda1a1c4a5f3)

mprahl committed:
fix(test): Fix the frontend sample test (#11968)

Signed-off-by: mprahl <mprahl@users.noreply.github.com>
(cherry picked from commit 54475637a241f55957149a32b80cb8c44d8f6458)

openshift-merge-bot[bot] committed:
Merge pull request #180 from mprahl/fix-samples-odh

Cherry-pick: fix(backend): Fix the pipeline samples (#11967)

dsp-developers commented:
Change to PR detected. A new PR build was completed.
  A set of new images have been built to help with testing out this PR: 
  **API Server**: `quay.io/opendatahub/ds-pipelines-api-server:pr-179`
  **DSP DRIVER**: `quay.io/opendatahub/ds-pipelines-driver:pr-179`
  **DSP LAUNCHER**: `quay.io/opendatahub/ds-pipelines-launcher:pr-179`
  **Persistence Agent**: `quay.io/opendatahub/ds-pipelines-persistenceagent:pr-179`
  **Scheduled Workflow Manager**: `quay.io/opendatahub/ds-pipelines-scheduledworkflow:pr-179`
  **MLMD Server**: `quay.io/opendatahub/mlmd-grpc-server:latest`
  **MLMD Envoy Proxy**: `registry.redhat.io/openshift-service-mesh/proxyv2-rhel8:2.3.9-2`
  **UI**: `quay.io/opendatahub/ds-pipelines-frontend:pr-179`
